libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Selectors/RandomSelector.cs

```diff
@@ -79,8 +79,11 @@ public Task<IReadOnlyList<int>> Select(SequenceContext context, CancellationToke
             var result = new List<int>();
             if (candidates.Count > 0)
             {
-                var selection = _rand.Next(candidates.Count);
-                result.Add(candidates[selection]);
+                lock (this)
+                {
```

Comment:
You technically only need the lock around the _rand.Next.  The result is local.

Reply:
Right, changed~